godot-core/src/builtin/plane.rs

```diff
@@ -191,6 +310,301 @@ mod test {
+        assert_eq_approx!(&plane_a, &plane_d, Plane::is_equal_approx);
+        // Although approximately equal in the `normal` part, it is not approximately equal in the `d`
+        // part.
+        assert_ne_approx!(&plane_a, &plane_e, Plane::is_equal_approx);
+        // Both attributes are approximately equal.
+        assert_eq_approx!(&plane_a, &plane_f, Plane::is_equal_approx);
+        // Although considered approximately equal with `plane_a`, `plane_b` is not considered approximately
+        // equal with `plane_e` because the baseline comparison is tighter.
+        assert_ne_approx!(&plane_b, &plane_e, Plane::is_equal_approx);
+    }
+
+    /// Tests `normalize()`.
+    #[test]
+    fn test_normalization() {
+        // Initial normalized planes (Denormalization not done here because the constructor panics
+        // when creating a plane with unnormalized planes).
+        let mut plane_a: Plane = Plane::new(Vector3::new(0.7, 2.0, 6.0).normalized(), 0.0);
+        let mut plane_b: Plane = Plane::new(Vector3::new(1.5, 7.2, 9.1).normalized(), 2.0);
+        let mut plane_c: Plane = Plane::new(Vector3::new(1.4, 9.1, 1.2).normalized(), 5.3);
+        let mut plane_d: Plane = Plane::new(Vector3::new(4.2, 2.9, 1.5).normalized(), 2.4);
+        let plane_e: Plane = Plane::new(Vector3::new(5.1, 3.0, 2.1).normalized(), 0.2);
+
+        // Denormalize planes.
+        plane_a.normal = Vector3::new(0.7, 2.0, 6.0);
+        plane_b.normal = Vector3::new(1.5, 7.2, 9.1);
+        plane_c.normal = Vector3::new(1.4, 9.1, 1.2);
+        plane_d.normal = Vector3::new(4.2, 2.9, 1.5);
+
+        // Verify normalization method.
+        assert_eq!(plane_a.normalized().normal, plane_a.normal.normalized());
+        assert_eq!(plane_b.normalized().normal, plane_b.normal.normalized());
+        assert_eq!(plane_c.normalized().normal, plane_c.normal.normalized());
+        assert_eq!(plane_d.normalized().normal, plane_d.normal.normalized());
+        assert_eq!(plane_e.normalized().normal, plane_e.normal.normalized());
+    }
+
+    /// Tests `is_finite()`.
+    #[test]
+    fn test_finite() {
+        // Initial ordinary planes.
+        let mut plane_a: Plane = Plane::new(Vector3::new(0.7, 2.0, -6.0).normalized(), 10.2);
+        let mut plane_b: Plane = Plane::new(Vector3::new(1.5, 7.2, 9.1).normalized(), 20.0);
+        let mut plane_c: Plane = Plane::new(Vector3::new(1.4, 9.1, 1.2).normalized(), 50.3);
+        let mut plane_d: Plane = Plane::new(Vector3::new(-4.2, 2.9, 1.5).normalized(), 20.4);
+        let plane_e: Plane = Plane::new(Vector3::new(7.2, -2.9, 2.2).normalized(), 3.3);
+
+        // Test with `normal`s.
+        plane_a.normal = Vector3::new(0.7, real::INFINITY, -6.0);
+        plane_b.normal = Vector3::new(0.7, 2.0, real::NEG_INFINITY);
+        plane_d.normal = Vector3::new(real::NAN, real::INFINITY, real::NEG_INFINITY);
+
+        // Test with `d`s.
+        plane_c.d = real::INFINITY;
+        plane_d.d = real::NAN;
```

Comment:
As above, directly initialize the planes with their final values.

It would even make sense to have initialization and test statements together, for clarity.
Just reuse the same variable name:
```rs
let plane = Plane { normal: Vector3::new(0.7, real::INFINITY, -6.0), d: 10.2 };
assert!(!plane.is_finite());

let plane = Plane { normal: Vector3::new(0.7, 2.0, real::NEG_INFINITY), d: 10.2 };
assert!(!plane.is_finite());

...
```